### PR TITLE
General fixes and improvements, server protocol page

### PR DESCRIPTION
**Pull request by juanjux**
The gRPC interface usage tutorial will be on the next PR (in the server protocol page).

### CODE REVIEW

File: architecture.md [hunk 1]
```diff
@@ -16,11 +19,20 @@ graph LR
 
 ## Language Drivers
 
-A language driver is a Docker container that takes file parsing requests and respond with the AST and the [UAST](./uast/specification.md).
+A language driver is a program, containerized with Docker, that takes file parsing
+requests with source code and reply with an [UAST](./uast/specification.md)
```
Comment (smola) [inline architecture.md]: `s/reply/replies`

File: driver/protocol.md [hunk 2]
```diff
@@ -34,14 +34,19 @@ Server-driver communication uses **protobuf 3** for message encoding. In the
 driver side, this is implemented in the UAST normalizer in Go, using the
 babelfish SDK.
 
-* **TODO: Add proto**
+You can see the messages used for the **protobuf 3** exchange on the [proto file
```
Comment (smola) [inline driver/protocol.md]: We are currently using JSON representation for server-driver communication as produced by Go's `encoding/json` package.

Comment (juanjux) [inline driver/protocol.md]: Fixed.

File: architecture.md (hunk 1 above)
Comment (juanjux) [inline architecture.md]: Fixed, good catch.

**Comment by juanjux** [comment]
I'm going to close and redo this PR (with the feedback incorporated) since I'm having problems rebasing with upstream/master so I prefer to start with a clean branch.

**Pull request closed without merging**